src/qibo/core/callbacks.py

```diff
@@ -50,6 +50,9 @@ def density_matrix(self, x):
     @abstract_callbacks.Callback.nqubits.setter
     def nqubits(self, n: int):
         from qibo import gates
+        if self._nqubits is not None and self._nqubits != n:
+            raise_error(RuntimeError,
+                        "Changing nqubits for EntanglementEntropy is not supported.")

```

Comment:
Perhaps we could make this error more explicit by printing the values of `n` and `self._nqubits` to show how they differ.